code/workhorse/screen_index_helper.R

```diff
@@ -0,0 +1,142 @@
+compute_covariance_h <- function(simulator,set1,set2){
+  #set2 can be pool or benchmark or the merge of these two
+  # set1= as.matrix(set1)
+  # set2 = as.matrix(set2)
+  
+  d = simulator$model[["numDim"]]
+  beta = simulator$model[["beta"]]
+  sigma = simulator$model[["sig2"]]
+  number = nrow(set1)
+  newcov=matrix(nrow=dim(set2)[1],ncol=number)
+  
+  for ( i in 1:dim(set2)[1]) {
```

Comment:
comment on the nested loop,  and perhaps use a function for the inner loop.

Reply:
The nested for loop has two levels. One level is for the row of a matrix and the other is for the column. In this case, we would keep the nested for loop as a whole rather than breaking it. The inner loop only has two lines, which are used only here, so I did not see the necessity to put them in a function.
```
  for ( i in 1:dim(set2)[1]) {
    for (j in 1:nrwo(set1)) {
      dist <- -sum((set2[i, 1:num_features]-set1[j, 1:num_features])^2*beta)
      cov_matrix[i, j] <- sigma*exp(dist)
    }   
  }
```